upload/src/addons/SV/WarningImprovements/XF/Service/User/WarningPointsChange.php

```diff
@@ -20,22 +20,85 @@ protected function applyWarningAction(WarningAction $action)
     {
         parent::applyWarningAction($action);
 
-        if (!empty(Globals::$warnngObj))
+        if (!empty(Globals::$warningObj))
         {
             if ((empty($this->lastWarningAction) || $action->points > $this->lastWarningAction->points) && (!empty($action->sv_post_node_id) || !empty($action->sv_post_thread_id)))
             {
-                $this->lastWarningAction = $action;
+                $this->lastAction = $action;
             }
         }
     }
 
+    protected function getActions()
+    {
+        $actions = null;
+
+        if (!empty(Globals::$warningObj))
+        {
+            /** @var \SV\WarningImprovements\Repository\WarningCategory $warningCategoryRepo */
+            $warningCategoryRepo = $this->repository('SV\WarningImprovements:WarningCategory');
+
+            if (Globals::$warningObj->warning_definition_id === 0)
```

Comment:
Yuck. This affects anything which tries to touch the Definition relation.

Let's abuse XF2's entity system;

Add a non-caching `getDefinition()` getter to the Warning entity, which tests to see if `warning_definition_id === 0`. It so it returns the Custom Warning Definition's definition, otherwise access `$this->Definition_` to get the actual definition (note the trailing underscope).

Reply:
@Xon done with this commit : https://github.com/Xon/XenForo2-WarningImprovements/pull/45/commits/0d4ce329666d2f1c142bd3e2525db0a28b752d8f

Reply:
I pushed some cleanup as the look up for the custom warning definition was done in another place, and there is an entity cache we can query